common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/TestShuffleDataContext.java

```diff
@@ -67,8 +67,10 @@ public void cleanup() {
   }
 
   /** Creates reducer blocks in a sort-based data format within our local dirs. */
-  public void insertSortShuffleData(int shuffleId, int mapId, byte[][] blocks) throws IOException {
+  public void insertSortShuffleData(
+      int shuffleId, int mapId, byte[][] blocks, boolean indeterminateBlock) throws IOException {
     String blockId = "shuffle_" + shuffleId + "_" + mapId + "_0";
+    if (indeterminateBlock) blockId += "_0";
```

Comment:
`_0` is a little confusing here. Maybe we should have a `shuffleGenerationId` parameter, instead of 
 `indeterminateBlock`.